src/huggingface_hub/hf_api.py

```diff
@@ -3542,6 +3542,38 @@ def update_repo_visibility(
         hf_raise_for_status(r)
         return r.json()
 
+    @validate_hf_hub_args
+    def update_repo_settings(
+        self,
+        repo_id: str,
+        gated: Union[str, bool] = False,
+        *,
+        token: Union[str, bool, None] = None,
+        repo_type: Optional[str] = None,
+    ) -> Dict[str, Union[str, bool]]:
+        if gated not in ["auto", "manual", False]:
+            raise ValueError("Invalid gated status, must be one of 'auto', 'manual', or False")
+        # Build headers
+        # headers = build_hf_headers(token=token)
+        r = get_session().put(
+            url=f"{self.endpoint}/api/datasets/{repo_id}/settings",
+            # url=f"https://huggingface.co/api/{repo_type}s/{repo_id}/settings",
+            headers=self._build_hf_headers(token=token),
+            json={"gated": gated},
+        )
+
+        try:
+            hf_raise_for_status(r)
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 401:
+                raise ValueError("Invalid or missing Hugging Face token. Please check your authentication.") from e
+            elif e.response.status_code == 404:
+                raise ValueError(f"Repository not found: {repo_id}") from e
+            else:
+                print(f"HTTP Error {e.response.status_code}: {e}")
```

Comment:
`hf_raise_for_status` already handles error codes like 401 or 404 with correct messages and exceptions so no need to re-implement it. In this case, you can simply do

```py
        hf_raise_for_status(r)
```
and that's enough. In general, it's good practice to raise the root error to the user (here a HTTPError), rather than a `ValueError`.